src/prepareData.js

```diff
@@ -18,7 +18,7 @@ export default function prepareData(renderProps, networkLayer) {
       environment,
       initialReadyState,
       queryAggregator,
-      render,
+      render: render(otherMiddleware),
```

Comment:
I was going to suggest this for clean code and keeping the current interface, but I'm concerned about performance issues with the arrow function recalling applyRouterMiddleware on every client side render...

```jsx
export default function(props, otherMiddleware = []) {
  return applyRouterMiddleware(...otherMiddleware, {
    renderRouterContext: (child, props) => (
      <IsomorphicRelayRouterContext {...props}>
        {child}
      </IsomorphicRelayRouterContext>
    ),

    renderRouteComponent: useRelay.renderRouteComponent,
  })(props);
}
```

```jsx
export default function prepareInitialRender(environment, renderProps, otherMiddleware) {
  const queryAggregator = new QueryAggregator(renderProps);

  return IsomorphicRelay.prepareInitialRender({
    environment,
    Container: queryAggregator,
    queryConfig: queryAggregator.queryConfig,
  }).then(({ initialReadyState }) => ({
    ...renderProps,
    environment,
    initialReadyState,
    queryAggregator,
    render: (props) => render(props, otherMiddleware),
  }));
}
```

So maybe instead this, even though I like the code syntax less?

```jsx
export default function(props, otherMiddleware = []) {
  const render = applyRouterMiddleware(...otherMiddleware, {
    renderRouterContext: (child, props) => (
      <IsomorphicRelayRouterContext {...props}>
        {child}
      </IsomorphicRelayRouterContext>
    ),

    renderRouteComponent: useRelay.renderRouteComponent,
  });

  return (!!props) ? render(props) : render;
}
```

```jsx
export default function prepareInitialRender(environment, renderProps, otherMiddleware) {
  const queryAggregator = new QueryAggregator(renderProps);

  return IsomorphicRelay.prepareInitialRender({
    environment,
    Container: queryAggregator,
    queryConfig: queryAggregator.queryConfig,
  }).then(({ initialReadyState }) => ({
    ...renderProps,
    environment,
    initialReadyState,
    queryAggregator,
    render: render(null, otherMiddleware),
  }));
}
```

And then just removing line 21 above completely in `prepareData`.

Let me know your thoughts and I'll update the PR.